### PR TITLE
Add connection manager

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is my [WIP] proposal for the connection manager. I'd like to have a first review on this.

Targets #542.

### CODE REVIEW

File: src/connection/config_file.c [hunk 1]
```diff
@@ -88,8 +88,8 @@ ws_connection_loadconf(void) {
         return -1;
     }
 
-    struct ws_connection_manager* proc;
-    proc = ws_connection_manager_new(fd, deserializer, NULL);
+    struct ws_connection_processor* proc;
+    proc = ws_connection_processor_new(fd, deserializer, NULL);
```
Comment (neithernut) [inline src/connection/config_file.c]: This should be done via the _real_ connection _manager_.
You'll obviously need a method to do this.

File: src/connection/manager.c [hunk 2]
```diff
@@ -25,5 +25,113 @@
...
+#include "connection/processor.h"
+#include "objects/set.h"
+#include "serialize/deserializer.h"
+#include "serialize/json/deserializer.h"
+#include "serialize/json/serializer.h"
+#include "serialize/serializer.h"
+#include "util/socket.h"
+
+//!< @todo no hardcoded socket?
+#define SOCK_NAME "waysome.sock"
```
Comment (neithernut) [inline src/connection/manager.c:42]: The socket name will _always_ be hardcoded (or, at least a default value will be).

**Comment by neithernut** [comment]
You are missing deinitialization of the manager.

File: src/connection/manager.c (hunk 2 above)
Comment (matthiasbeyer) [inline src/connection/manager.c:42]: I will remove the `@todo`.

File: src/connection/config_file.c (hunk 1 above)
Comment (matthiasbeyer) [inline src/connection/config_file.c]: What exactly do you mean by "this"? The whole configuration loading?

Comment (neithernut) [inline src/connection/config_file.c]: The contruction of a connection for the config.

Comment (matthiasbeyer) [inline src/connection/config_file.c]: You would add this to the scope of this PR? I would consider this as stuff for a second PR, as it is mostly refactoring work...

Comment (neithernut) [inline src/connection/config_file.c]: Nope, you just have to exrchange a function call or two.

**Comment by matthiasbeyer** [comment]
@neithernut Please review if this is what you wanted me to do (the latest 2 commits).

File: src/connection/config_file.c [hunk 3]
```diff
@@ -76,25 +76,17 @@ open_config(void);
 
 int
 ws_connection_loadconf(void) {
-    int fd = open_config();
+    int fd = ws_connection_manager_init();
```
Comment (neithernut) [inline src/connection/config_file.c]: Please, at least rename `fd` to `res`.

**Comment by neithernut** [comment]
Although the config manager is initialized on config load, you should probably initialize it explicitely in `main()` for good meassure.

**Comment by matthiasbeyer** [comment]
You mean the connection manager, but yes.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**